python/server.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from string import Template
+from math import floor
+import os
+import hmac
+import hashlib
+import time
+
+# Create custom HTTPRequestHandler class
+class EmbeddedChartHTTPRequestHandler(BaseHTTPRequestHandler):
+
+    def get_embedding_url(self, chartId):
+        # remember, in a real application, you should ensure your user is authenticated and authorized to view the chart before you return the signed URL.
+
+        # replace these constants with the correct values for your Charts instance
+        CHARTS_EMBEDDING_BASE_URL = '~REPLACE~CHARTS_EMBEDDING_BASE_URL'  # Replace with the base URL to your Charts instance, e.g. https://charts.mongodb.com/charts-foo-abcde
+        CHARTS_TENANT_ID = '~REPLACE~CHARTS_TENANT_ID'  # Replace with your Charts Tenant ID from the Embed Chart snippet
+        EMBEDDING_SIGNING_KEY = '~REPLACE~EMBEDDING_SIGNING_KEY'  # Replace with the Embedding Signing Key generated by your Charts admin
+        EXPIRY_TIME_SECONDS = 300  # Set to your preferred expiry period
+
+        timestamp = int(floor(time.time()))
+        payload = Template('id=$chartId&tenant=$tenantId&timestamp=$timestamp&expires-in=$expiry').substitute(chartId=chartId, tenantId=CHARTS_TENANT_ID, timestamp=timestamp, expiry=EXPIRY_TIME_SECONDS)
+        signature = hmac.new(EMBEDDING_SIGNING_KEY.encode('UTF-8'), msg=payload.encode('UTF-8'), digestmod=hashlib.sha256).hexdigest()
+        url = Template('$baseUrl/embed/charts?$payload&signature=$signature').substitute(baseUrl=CHARTS_EMBEDDING_BASE_URL, payload=payload, signature=signature)
+
+        return url
+
+    # helper method to return 200 response header
+    def send_headers(self):
+        self.send_response(200)
+        self.send_header('Content-type','text/html')
+        self.end_headers()
+
+    # handle GET command
+    def do_GET(self):
+        # serve embedding signature
+        if self.path.startswith('/api/embeddedchart/'):
+            self.send_headers()
+            # TODO: parse chart ID from path
+            self.wfile.write(bytes(self.get_embedding_url('51efab7a-3b1f-4fa9-82db-8c31b83b3df9'), 'UTF-8'))
```

Comment:
This TODO needs to be done!

Reply:
strange, I definitely implemented that. Perhaps I made a mistake squashing the commits and it got lost. Will re-implement :-)